### DISCUSSION TITLE
How can a custom button be used to switch the brush type to circle, arrow, line, etc.? I haven't found a method in JavaScript to control the brush type switch.

### DISCUSSION THREAD

**Discussion by PineSongCN** [personalizedrefrigerator/js-draw#82] [Q&A]
How can a custom button be used to switch the brush type to circle, arrow, line, etc.? I haven't found a method in JavaScript to control the brush type switch.

**Comment by personalizedrefrigerator** [comment]
This can be done with `.setStrokeFactory`. For example, to switch to a circle:
```ts
import { PenTool, Editor, makeOutlinedCircleBuilder } from 'js-draw';

const editor = new Editor(document.body);

const firstPen = editor.toolController.getMatchingTools(PenTool)[0];
firstPen.setStrokeFactory(makeOutlinedCircleBuilder);
```

There isn't currently a way to get the list of default brush types.

See also:
- The example just above ["Custom pens" in the "Customizing tools" documentation](https://personalizedrefrigerator.github.io/js-draw/typedoc/modules/Additional_Documentation.CustomizingTools__.html#md:custom-pens).

**Reply by PineSongCN** [reply]
Sorry, I meant that I customized a few buttons outside the editor, and changed the editor's brush type after clicking the buttons

**Comment by PineSongCN** [comment]
@personalizedrefrigerator  I meant that I customized a few buttons outside the editor, and changed the editor's brush type after clicking the buttons.
